test/tests/functional/pbs_hook_perf_stat.py

```diff
@@ -0,0 +1,482 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2019 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+
+from tests.functional import *
+
+
+class Test_hook_perf_stat(TestFunctional):
+    """
+    This test suite tests that the hook performance stats are in place
+    """
+
+    def setUp(self):
+        TestFunctional.setUp(self)
+        # ensure LOG_EVENT_DEBUG3 is being recorded to see perf stats
+        a = {'log_events': 4095}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        self.mom.add_config({'$logevent': 4095})
+        self.mom.signal('-HUP')
+
+    def test_queuejob_hook(self):
+        """
+        Test that pbs_server collects performance stats for queuejob hook
+        """
+        hook_content = """
```

Comment:
The hook content is repeated multiple times in different tests, maybe add it once under the class?

Reply:
Good idea. I'll update.